refinery/data_set_manager/views.py

```diff
@@ -1170,7 +1181,41 @@ def patch(self, request, uuid):
                 return Response(
                     NodeSerializer(node).data, status=status.HTTP_200_OK
                 )
-            return Response('Currently, you can only remove node files.',
+            # derived node can have multiple attribute sources
+            elif solr_name and attribute_value and not node.is_derived():
+                # splits solr name into type and subtype
+                attribute_obj = customize_attribute_response([solr_name])[0]
+                annotated_nodes_query = AnnotatedNode.objects.filter(
+                    node=node,
+                    attribute_type=attribute_obj.get('attribute_type'),
+                    attribute_subtype__icontains=attribute_obj.get(
+                        'display_name'
+                    )
+                )
+
+                # update the source attribute
+                try:
+                    source_attribute = annotated_nodes_query[0].attribute
```

Comment:
Do we know that the first item in this queryset is the AnnotatedNode that we want? Also, remember that the order of the `filter()` isn't guaranteed.

Reply:
Yes, it should only be one item unless there duplicates. If there duplicates, it doesn't matter as they will be updated below as long as the source attribute gets updated. Of course, if you can think of user edge cases you'd like for me to add as a unit test, please let me know.